packages/clay-form/src/Form.tsx

```diff
@@ -103,15 +103,21 @@ const FeedbackIndicator = React.forwardRef<
 	)
 );
 
-const ClayForm: React.FunctionComponent<
-	React.HTMLAttributes<HTMLFormElement>
+type Form = React.ForwardRefExoticComponent<
+	React.HTMLAttributes<HTMLFormElement> & React.RefAttributes<HTMLFormElement>
 > & {
 	FeedbackGroup: typeof FeedbackGroup;
 	FeedbackIndicator: typeof FeedbackIndicator;
 	FeedbackItem: typeof FeedbackItem;
 	Group: typeof Group;
 	Text: typeof Text;
-} = ({children, ...otherProps}) => <form {...otherProps}>{children}</form>;
+};
+
+const ClayForm = React.forwardRef(({children, ...otherProps}, ref) => (
+	<form {...otherProps} ref={ref}>
+		{children}
+	</form>
+)) as Form;
```

Comment:
Instead of using `as ...`, I think we could start using `Object.assign()` like i did here in this PR, https://github.com/liferay/clay/pull/2700/commits/5af7b9dd5163f7f7351c24fe57317b88028f2fd6#diff-c6dc8bae2d26a6f2f689a012da29b157.

This will help react-docgen, since they currently dont support `as`